index.js

```diff
@@ -227,6 +228,68 @@ function inject (bot) {
     if (Math.abs(bot.entity.position.z - blockZ) > 0.2) { bot.entity.position.z = blockZ }
   }
 
+  function moveToEdge (pos, edge) {
+    const dirVector = {
+      '1,0': new Vec3(-0.540, -0.841, 0),
+      '-1,0': new Vec3(0.540, -0.841, 0),
+      '0,1': new Vec3(0, -0.841, -0.540),
+      '0,-1': new Vec3(0, -0.841, 0.540)
+    }
+    const anglesEuler = {
+      '1,0': { yaw: 1.570, pitch: -1 },
+      '-1,0': { yaw: -1.570, pitch: -1 },
+      '0,1': { yaw: 0, pitch: -1 },
+      '0,-1': { yaw: 3.141, pitch: -1 }
+    }
+    // If allowed turn instantly should maybe be an option
+    const allowInstantTurn = false
+    // Target viewing direction
+    const yaw = anglesEuler[parseInt(edge.x) + ',' + parseInt(edge.z)].yaw
+    const pitch = -1
+    // Figure out if the bot is looking the right way
+    function getViewDirection (pitch, yaw) {
+      const csPitch = Math.cos(pitch)
+      const snPitch = Math.sin(pitch)
+      const csYaw = Math.cos(yaw)
+      const snYaw = Math.sin(yaw)
+      return new Vec3(-snYaw * csPitch, snPitch, -csYaw * csPitch)
+    }
+    const view = getViewDirection(bot.entity.pitch, bot.entity.yaw)
+    const idealDir = dirVector[parseInt(edge.x) + ',' + parseInt(edge.z)]
+    // Check the distance it is away from the target position
+    if (bot.entity.position.distanceTo(pos.offset(edge.x + 0.5, 1, edge.z + 0.5)) > 0.6) {
+      // Check if the Bot is still turning to the target view
+      if (view.dot(idealDir) < 0.9) {
+        bot.look(yaw, pitch, allowInstantTurn, () => {})
+        bot.setControlState('sneak', true)
+        return false
+      }
+      bot.setControlState('sneak', true)
+      bot.setControlState('back', true)
+      return false
+    }
+    bot.setControlState('back', false)
+    return true
+  }
+
+  function moveToBlock (block) {
+    // TODO should be replaced with sneak when it is implemented in prismarine-physics
+    const maxMovement = 0.1
+    const targetPos = block.clone().offset(0.5, 0, 0.5)
+    console.log(targetPos)
+    if (bot.entity.position.distanceTo(targetPos) > 0.001) {
+      const targetVec = targetPos.clone().subtract(bot.entity.position).normalize()
+      if (maxMovement * maxMovement < bot.entity.position.distanceSquared(targetPos)) {
+        targetVec.scale(maxMovement)
+      } else {
+        targetVec.scale(bot.entity.position.distanceTo(targetPos))
+      }
+      bot.entity.position.add(targetVec)
```

Comment:
It would be nice to not edit the position directly.

Reply:
Yes the problem is that without editing the position directly I would have to tell the bot somehow to correct his position by itself and I don't know how to do that exactly. The position is to precise to be adjusted by mineflayer-pathfinder as far as I know.